packages/astro/src/core/errors/errors.ts

```diff
@@ -141,6 +141,12 @@ export class AggregateError extends AstroError {
 	}
 }
 
+/**
+ * Check if an error is a ZodError from an AstroConfig validation.
+ * Used to suppress formatting a ZodError if needed.
+ */
+export const astroConfigZodErrors = new WeakSet();
```

Comment:
Usually, I tend to **not expose** such internal types from a module. I tend to expose functions that internally mutate the object. Nothing against it, but it's a pattern that I don't feel comfortable about, it seems more a side effect.

Reply:
Yeah I figured it could be a function too, but then I'd have to export two `setAstroConfigZodError` and `isAstroConfigZodError` APIs instead of one. I'm leaning towards not putting it in `config.ts` to remove the need for `setAstroConfigZodError` because that leads to the CLI `throw-and-exit.ts` needing to take longer to be `import()`-ed.

Reply:
I pushed the changes. Admittedly I think it makes things more cleaner now (had type issues) so thanks for nudging me to that 😄